File: itc/files/C16/terminaison.py
# Renvoie le quotient dans la division euclidienne de a par b avec a et b deux entiers naturels et b non nul
def quotient(a, b):
    assert (a >= 0 and b > 0)
    q = 0
    while (a - b >= 0):
        a = a - b
        q = q + 1
    return q

# Renvoie True si elt est dans liste et False sinon
def est_dans(elt, liste):
    if liste == []:
        return False
    return elt == liste[0] or est_dans(elt, liste[1:])


def syracuse(n):
    while n != 1:
        if n % 2 == 0:
            n = n//2
        else:
            n = 3*n+1
    return True

File: mp2i/files/C7/terminaison.c
#include <stdio.h>
#include <assert.h>

/* Renvoie le quotient dans la division euclidienne de a par b avec a et b deux entiers naturels et b non nul  */
int quotient(int a,  int b)
{   assert (a>=0 && b>0);
    int q = 0;
    while (a - b >= 0)
    {
        a = a - b;
        q = q + 1;
    }
    return q;
}

int main()
{
    printf("Quotient de 2025 par 15 = %d\n",quotient(2025,15));
}